tools/codegen/model.py

```diff
@@ -1452,11 +1452,19 @@ def __post_init__(self) -> None:
                     f"variant, {str(f.native_function.func.name)} will be generated for you " \
                     "and doesn't need to live in the yaml."
 
-    def functions(self) -> Iterator[ExternalBackendFunction]:
-        yield self.out
-        yield self.functional
-        if self.inplace is not None:
-            yield self.inplace
+    def functions(self, *, functional_first: bool = False) -> Iterator[ExternalBackendFunction]:
+        if not functional_first:
+            yield self.out
+            yield self.functional
+            if self.inplace is not None:
+                yield self.inplace
+        else:
+            # When we generate out/inplace wrappers for unstructured kernels, we need the functional kernel
+            # to be defined before we can generate the other two wrappers.
+            yield self.functional
+            yield self.out
+            if self.inplace is not None:
+                yield self.inplace
```

Comment:
so... when do you ever need the out variant to come first?

Reply:
yeah you don't. I guess I just left the `out`-first version in to be consistent with in-tree codegen, but that's not really necessary. I'll just kill the flag and add a comment explaining why we need functional to come first.